frontend/src/scenes/trends/ActionFilter/ActionFilter.js

```diff
@@ -1,33 +1,77 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useState } from 'react'
 import { useActions, useValues } from 'kea'
+import GridLayout, { WidthProvider } from '@mariusandra/react-grid-layout'
 import { entityFilterLogic } from './entityFilterLogic'
 import { ActionFilterRow } from './ActionFilterRow'
 import { Button } from 'antd'
 
+const ReactGridLayout = WidthProvider(GridLayout)
+
 export function ActionFilter({ setFilters, filters, typeKey, hideMathSelector }) {
     const logic = entityFilterLogic({ setFilters, filters, typeKey })
 
-    const { localFilters } = useValues(logic)
-    const { addFilter, setLocalFilters } = useActions(logic)
+    const { localFilters, layouts } = useValues(logic)
+    const { addFilter, orderFilters, setLocalFilters } = useActions(logic)
+    // This is to be able to collapse a row when it is being dragged due to a bug
+    // with reac-grid-layout that doesn't seems to work when a row is open
+    const [draggedRow, setDraggedRow] = useState(null)
 
     // No way around this. Somehow the ordering of the logic calling each other causes stale "localFilters"
     // to be shown on the /funnels page, even if we try to use a selector with props to hydrate it
     useEffect(() => {
         setLocalFilters(filters)
     }, [filters])
 
+    const renderLocalFilters = () =>
+        localFilters.map((filter, index) => (
+            <div key={filter.id.toString()} data-grid={layouts.find((layout) => layout.i === filter.id)}>
+                <ActionFilterRow
+                    logic={logic}
+                    filter={filter}
+                    index={index}
+                    key={index}
+                    hideMathSelector={hideMathSelector}
+                    dragging={draggedRow === filter.id}
+                />
+            </div>
+        ))
+    // eslint-disable-next-line no-unused-vars
+    const updateFilterPositions = (layout, _oldItem, _newItem) => {
+        setDraggedRow(null)
+        const filterPositions = layout.reduce((positions, filter) => {
+            positions[filter.i] = filter.y
+            return positions
+        }, {})
+        orderFilters(filterPositions)
+    }
+
+    // eslint-disable-next-line no-unused-vars
+    const collapseFilter = (_layout, _oldItem, newItem) => {
+        setDraggedRow(Number(newItem.i))
+    }
+
     return (
-        <div>
-            {localFilters &&
-                localFilters.map((filter, index) => (
-                    <ActionFilterRow
-                        logic={logic}
-                        filter={filter}
-                        index={index}
-                        key={index}
-                        hideMathSelector={hideMathSelector}
-                    />
-                ))}
+        <div style={{ position: 'relative' }}>
+            {localFilters && (
+                <ReactGridLayout
```

Comment:
The problem is that this component is also used in /trends, which doesn't need to be re-ordered but in this case is also crashing

Reply:
Just thinking about it, I think it's a good thing to be able to re-order the actions on /trends too, the only thing we need to do is make sure it doesn't crash the trends page :)

Reply:
Oops. Didn't notice that 😅 . Let me fix it

Reply:
It should be fixed now. I was able to add and reorder filters at ease in both pages.
![add_filter](https://user-images.githubusercontent.com/15788901/88235585-8d1b1100-cc51-11ea-864a-5fcad592613e.png)